Runtime/MeshCombiner.cs

```diff
@@ -516,6 +516,46 @@ private static Matrix4x4 ScaleMatrix(ref Matrix4x4 matrix, float scale)
                 m33 = matrix.m33 * scale
             };
         }
+
+        static bool CanReadMesh(Mesh mesh) {
+            #if UNITY_EDITOR
+            return CanReadMeshInEditor(mesh);
+            #else
+            return mesh.isReadable;
+            #endif
+        }
+
+#if UNITY_EDITOR
+        static System.Reflection.MethodInfo Mesh_canAccess;
+        // This is a workaround for a Unity peculiarity - 
+        // non-readable meshes are actually always accessible from the Editor.
+        // We're still logging a warning since this won't work in a build.
+        static bool CanReadMeshInEditor(Mesh mesh)
+        {
+            if (mesh.isReadable) return true;
+            if (!Application.isPlaying) return true;
+
+            if (Mesh_canAccess == null)
+                Mesh_canAccess = typeof(Mesh).GetProperty("canAccess", System.Reflection.BindingFlags.NonPublic | System.Reflection.BindingFlags.Instance).GetMethod;
+
+            if(Mesh_canAccess != null)
+            {
+                try {
+                    bool canAccess = (bool) Mesh_canAccess.Invoke(mesh, null);
+                    if(canAccess) {
+                        Debug.LogWarning("The mesh you are trying to export is not marked as readable. This will only work in the Editor and fail in a Build.", mesh);
```

Comment:
Could you perhaps replace **export** with **access**" so that it's more clear? Since this is not used to export anything.